src/views/LoginSuccess.vue

```diff
@@ -61,142 +63,116 @@ export default class LoginSuccess extends Vue {
-            userFriendlyAddresses.forEach((addr) => {
-                const addressInfo = this.accounts.get(addr);
-                addressInfo!.balance = Nimiq.Policy.coinsToSatoshis(balances.get(addr) || 0);
-                this.accounts.set(addr, addressInfo!);
-            });
-            this.accountsUpdateCount += 1;
-
-            // Kick off account detection
-            this.findAccounts();
+            || this.keyguardResult.keyType === WalletType.LEDGER) {
+            // read accounts from network
+            const foundAccounts = await AccountFinder.findAccounts(
+                this._deriveKeyguardAccounts.bind(this),
+                () => Promise.resolve(this.keyguardResult.keyId),
+                LoginSuccess.DEFAULT_ACCOUNT_LABEL,
+                this._addAccounts.bind(this),
+            );
+            this._addAccounts(foundAccounts);
         }
-    }
 
-    private async findAccounts() {
-        // The standard dictates that account detection terminates
-        // when 20 consecutive unused accounts have been found.
-        if (this.lastDerivedIndex >= this.lastActiveIndex + 20) {
-            // End condition
+        if (this.keyguard) {
             this.keyguard.releaseKey(this.keyguardResult.keyId);
-            // TODO network visuals with longer than 1 list of accounts during retrieval
-            this.retrievalComplete = true;
-            return;
         }
+        // TODO network visuals with longer than 1 list of accounts during retrieval
+        this.retrievalComplete = true;
+    }
 
-        // 1. Generate paths to derive
-
-        // To be able to use the lastDerivedIndex as the index for the pathsToDerive array,
-        // we need to fill it up until (and including) the lastDerivedIndex with `empty` values.
-        const pathsToDerive: string[] = new Array(this.lastDerivedIndex + 1);
-
-        // We can then push the paths of the relevant accounts onto the end of the array
-        for (let i = this.lastDerivedIndex + 1; i <= this.lastActiveIndex + 20; i++) {
-            pathsToDerive.push(`m/44'/242'/0'/${i}'`);
+    private async _setInitialAccounts() {
+        // read potentially existing wallet and accounts from WalletStore
+        const currentWalletInfo = await WalletStore.Instance.get(this.keyguardResult.keyId);
+        if (currentWalletInfo) {
+            this.walletLabel = currentWalletInfo.label || this.walletLabel;
+            if (currentWalletInfo.accounts) {
+                this.accounts = currentWalletInfo.accounts;
+            }
         }
 
-        // 2. Derive accounts from paths
+        // The Keyguard always returns (at least) one derived Address,
+        // which we'll also already add and display
+        const keyguardResultAccounts = this.keyguardResult.addresses.map((addressObj) => ({
+            address: new Nimiq.Address(addressObj.address).toUserFriendlyAddress(),
+            keyPath: addressObj.keyPath,
```

Comment:
It's just a code simplification. `_addAccounts` is generalized to potentially update the balance which does not apply here, but still just calling the same method makes the code shorter and more readable.

I'm using keyPath, because it's called like this in the KeyguardResult and therefore also in the LedgerApi result.

Reply:
Ok, so the hardware and software wallets return it, fine, but in the Accounts Manager we use `path`.